### DISCUSSION TITLE
Image guidance for DS

### DISCUSSION THREAD

**Discussion by zarahzachz** [CMSgov/design-system#2226] [Ideas]
Conversation came up in [Slack](https://adhoc.slack.com/archives/C645GLLNN/p1666883557550989).

Questions:
- Shouldn’t sites have image compression
    - CI/CD actions?
    - Akamai/imgix?
- Would an image component help?
    - Built-in skeleton (loading) state
    - [`lazy-loading` attr](https://web.dev/browser-level-image-lazy-loading/) baked in
- [Intersection Observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)
   - Detail use?
   - Distribute some kind of reusable code? Might prove difficult for page layouts as DS covers more granular patterns

**Comment by zarahzachz** [comment]
Reminds me of the post [GH wrote on releasing performant images](https://github.blog/2021-01-29-making-githubs-new-homepage-fast-and-performant/)

**Comment by danbivins** [comment]
Adding a "performance" section on the doc site could be useful to cover various topics.

Is there any downside to creating an image component?

**Reply by zarahzachz** [reply]
One I can see is if we introduce skeleton (or loading) states in our components, it opens the floodgates to other components also needing this loading effect. Skeleton states tend to be animation heavy based on how they are implemented, which can lead to performance issues or excessive motion on the website. All workable issues, but worth calling out here.
